routing/cross_mwm_connector_serialization.hpp

```diff
@@ -387,10 +440,20 @@ class CrossMwmConnectorSerializer final
     buffer.clear();
   }
 
-  static void WriteTransitions(std::vector<Transition> const & transitions,
+  template <typename CrossMwmId>
+  static void WriteTransitions(std::vector<Transition<CrossMwmId>> const & transitions,
                                serial::CodingParams const & codingParams, uint32_t bitsPerOsmId,
-                               uint8_t bitsPerMask, std::vector<uint8_t> & buffer);
+                               uint8_t bitsPerMask, std::vector<uint8_t> & buffer)
+  {
+    MemWriter<vector<uint8_t>> memWriter(buffer);
+
+    for (auto const & transition : transitions)
+      transition.Serialize(codingParams, bitsPerOsmId, bitsPerMask, memWriter);
+  }
 
   static void WriteWeights(std::vector<Weight> const & weights, std::vector<uint8_t> & buffer);
 };
+
+static_assert(sizeof(transit::StopId) * 8 == 64, "Wrong transit::StopId size");
```

Comment:
static_assert(CHAR_BIT == 8, "");

Reply:
Could you please explain what you mean? CHAR_BIT is used in routing/road_access_serialization.hpp

Reply:
Why do you use 8 here? Because of eight bits in a byte? But there exists archs with 9 or 7 bits in a byte, and CHAR_BIT is a number of bits in a byte.

Reply:
The idea behind the static_assert() was. There're methods for writing and reading TransitId:
```c++
    template <class Sink>
    void WriteCrossMwmId(connector::TransitId const & id, uint8_t /* n */, BitWriter<Sink> & w) const
    {
      w.WriteAtMost64Bits(id.m_stop1Id, kStopIdBits);
      w.WriteAtMost64Bits(id.m_stop2Id, kStopIdBits);
      w.WriteAtMost32Bits(id.m_lineId, kLineIdBits);
    }
...
    template <class Source>
    void ReadCrossMwmId(uint8_t /* n */, BitReader<Source> & reader, connector::TransitId & readed)
    {
      readed.m_stop1Id = reader.ReadAtMost64Bits(kStopIdBits);
      readed.m_stop2Id = reader.ReadAtMost64Bits(kStopIdBits);
      readed.m_lineId = reader.ReadAtMost32Bits(kLineIdBits);
    };
```
Methods with number 64 and 32 and kStopIdBits(64) and kLineIdBits()32 are used. I want to to here if somebody changes types for stop id (uint64_t) and line id(uint32_t) the compilation will be broken.

How about:
```
static_assert(std::is_same<transit::StopId, uint64_t>::value, "Wrong transit::StopId type.");
static_assert(std::is_same<transit::LineId, uint32_t>::value, "Wrong transit::LineId type.");
```

Reply:
Better is:
```c++
static_assert(kStopIdBits <= 64, "");
static_assert(kLineIdBits <= 32, "");
```

Reply:
Done